proposals/nnnn-bitwise-copyable.md

```diff
@@ -0,0 +1,270 @@
+
+## Future Directions
+
+### Automatic derivation of conditional conformances
+
+The wrapper type mentioned above
+```swift
+struct RegularBox<Value> {
+  var first: Value
+}
+```
+cannot conform to `BitwiseCopyable` unconditionally.
+It can, however, so long as `Value` is `BitwiseCopyable`.
+
+With this proposal, such a conditional conformance can be added manually:
+
+```swift
+extension Box : BitwiseCopyable where Value : BitwiseCopyable {}
+```
+
+In the future we may in some cases be able to derive it automatically.
+
+### MemoryLayout<T>.isBitwiseCopyable
+
+In certain circumstances, it would be useful to be able to dynamically determine whether a type conforms to `BitwiseCopyable`.
+In order to allow that, a new field could be added to `MemoryLayout`.
+
+### BitwiseMovable
+
+Most Swift types have the property that their representation can be relocated in memory with direct memory operations.
+This could be represented with a `BitwiseMovable` protocol that would be handled similarly to `BitwiseCopyable`.
+
+## Alternatives considered
+
+### Alternate Spellings
+
+**Trivial** is widely used within the compiler and Swift evolution discussions to refer to the property of bitwise copyability. `BitwiseCopyable`, on the other hand, is more self-documenting.
+
+## Acknowledgments
+
+This proposal has benefitted from discussions with John McCall, Joe Groff, Andrew Trick, Michael Gottesman, and Arnold Schwaigofer.
+
+## Appendix: Standard library conformers<a name="all-stdlib-conformers"/>
+
+The following protocols in the standard library will gain the `BitwiseCopyable` constraint:
+
+- `FixedWidthInteger`
+- `_Pointer`
+- `SIMDStorage`, `SIMDScalar`, `SIMD`
+
+
+The following types in the standard library will gain the `BitwiseCopyable` constraint:
+
+- `Optional<T>` when `T` is `BitwiseCopyable`
+- The fixed-precision integer types:
+  - `Bool`
+  - `Int8`, `Int16`, `Int32`, `Int64`, `Int`
+  - `UInt8`, `UInt16`, `UInt32`, `UInt64`, `UInt`
+  - `StaticBigInt`
+  - `UInt8.Words`, `UInt16.Words`, `UInt32.Words`, `UInt64.Words`, `UInt.Words`
+  - `Int8.Words`, `Int16.Words`, `Int32.Words`, `Int64.Words`, `Int.Words`
+- The fixed-precision floating-point types:
+  - `Float`, `Double`, `Float80`
+  - `FloatingPointSign`, `FloatingPointClassification`
+- The family of `SIMDx<Scalar>` types
+- The family of unmanaged pointer types:
+  - `OpaquePointer`
+  - `UnsafeRawPointer`, `UnsafeMutableRawPointer`
+  - `UnsafePointer`, `UnsafeMutablePointer`, `AutoreleasingUnsafeMutablePointer`
+  - `UnsafeBufferPointer`, `UnsafeMutableBufferPointer`
+  - `UnsafeRawBufferPointer`, `UnsafeMutableRawBufferPointer`
+  - `Unmanaged`
+  - `CVaListPointer`
+- Some types related to collections
+  - `EmptyCollection`
+  - `UnsafeBufferPointer.Iterator`, `UnsafeRawBufferPointer.Iterator`, `EmptyCollection.Iterator`
+  - `String.Index`, `CollectionDifference.Index`
+- Some types related to unicode
+  - `Unicode`
```

Comment:
We considered some of the uninhabited types here and we feel that we need to be more selective about which get the conformance, even though they're trivially bitwise copyable by virtue of being uninhabited and thus zero-sized. Rather than looking at just the size of the type, should we consider intent—as in, would a user reasonably write a generic algorithm constrained by `BitwiseCopyable` that might involve one of these types?

We decided that the answer is probably **no** for `Unicode`, `MemoryLayout`, and `CommandLine`, as those are purely namespaces for other declarations. But on the flipside, the answer is probably **yes** for `Never` and `SystemRandomNumberGenerator`; `Never` does occur in generic contexts and we have precedence for adding other conformances to it for that purpose. Likewise, despite being zero-size, users do create values of `SystemRandomNumberGenerator` so the conformance makes sense there.

Reply:
Updated the implementation and proposal accordingly.  And added a few words to describe the inference suppression mechanism.